connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/PluginScanResult.java

```diff
@@ -27,32 +27,35 @@
 import org.apache.kafka.connect.transforms.predicates.Predicate;
 
 import java.util.Arrays;
-import java.util.Collection;
+import java.util.SortedSet;
 import java.util.List;
+import java.util.TreeSet;
+import java.util.function.Consumer;
+import java.util.function.Function;
 
 public class PluginScanResult {
-    private final Collection<PluginDesc<SinkConnector>> sinkConnectors;
-    private final Collection<PluginDesc<SourceConnector>> sourceConnectors;
-    private final Collection<PluginDesc<Converter>> converters;
-    private final Collection<PluginDesc<HeaderConverter>> headerConverters;
-    private final Collection<PluginDesc<Transformation<?>>> transformations;
-    private final Collection<PluginDesc<Predicate<?>>> predicates;
-    private final Collection<PluginDesc<ConfigProvider>> configProviders;
-    private final Collection<PluginDesc<ConnectRestExtension>> restExtensions;
-    private final Collection<PluginDesc<ConnectorClientConfigOverridePolicy>> connectorClientConfigPolicies;
-
-    private final List<Collection<?>> allPlugins;
+    private final SortedSet<PluginDesc<SinkConnector>> sinkConnectors;
+    private final SortedSet<PluginDesc<SourceConnector>> sourceConnectors;
+    private final SortedSet<PluginDesc<Converter>> converters;
+    private final SortedSet<PluginDesc<HeaderConverter>> headerConverters;
+    private final SortedSet<PluginDesc<Transformation<?>>> transformations;
+    private final SortedSet<PluginDesc<Predicate<?>>> predicates;
+    private final SortedSet<PluginDesc<ConfigProvider>> configProviders;
+    private final SortedSet<PluginDesc<ConnectRestExtension>> restExtensions;
+    private final SortedSet<PluginDesc<ConnectorClientConfigOverridePolicy>> connectorClientConfigPolicies;
+
+    private final List<SortedSet<?>> allPlugins;
```

Comment:
We can be more specific here, which simplifies the `PluginScanResult::forEach` implementation below:
```suggestion
    private final List<SortedSet<? extends PluginDesc<?>>> allPlugins;
```